pkg/controller/operators/operatorconditiongenerator_controller.go

```diff
@@ -0,0 +1,203 @@
+
+	in := &operatorsv1.Operator{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, in)
+	if err != nil {
+		log.V(1).Info("Unable to find operator", "error", err)
+		return ctrl.Result{}, err
+	}
+
+	// Wrap with convenience decorator
+	operator, err := r.factory.NewOperator(in)
+	if err != nil {
+		log.Error(err, "Could not wrap Operator with convenience decorator")
+		return reconcile.Result{Requeue: true}, nil
+	}
+
+	selector, err := operator.ComponentSelector()
+	if err != nil {
+		return reconcile.Result{Requeue: true}, nil
+	}
+	operatorConditionList := operatorsv1.OperatorConditionList{}
+
+	err = r.Client.List(context.TODO(), &operatorConditionList, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if len(operatorConditionList.Items) != 0 {
+		return ctrl.Result{}, nil
+	}
+	r.log.V(1).Info("Creating OperatorCondition")
+
+	// Get CSV Namespace
+	csvList := operatorsv1alpha1.ClusterServiceVersionList{}
+	err = r.Client.List(context.TODO(), &csvList, &client.ListOptions{LabelSelector: selector})
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// If you cannot identify the CSV for the operator do not proceed.
+	if len(csvList.Items) != 1 {
+		return ctrl.Result{Requeue: true}, nil
+	}
+
+	csv := csvList.Items[0]
+
+	labelKey, err := operator.ComponentLabelKey()
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// get all Service Accounts
+	serviceAccountList := corev1.ServiceAccountList{}
+	err = r.Client.List(context.TODO(), &serviceAccountList, client.InNamespace(csv.GetNamespace()))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	serviceAccounts := []string{}
+	for _, serviceAccount := range filterServiceAccounts(serviceAccountList.Items, hasOwnerFunc(ownerutil.NonBlockingOwner(&csv))) {
+		serviceAccounts = append(serviceAccounts, serviceAccount.GetName())
+	}
+	if len(serviceAccounts) == 0 {
+		serviceAccounts = []string{"default"}
+	}
+
+	// get all Service Accounts
+	deploymentList := appsv1.DeploymentList{}
+	err = r.Client.List(context.TODO(), &deploymentList, client.InNamespace(csv.GetNamespace()))
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	deployments := []string{}
+	for _, deployment := range filterDeployments(deploymentList.Items, hasOwnerFunc(ownerutil.NonBlockingOwner(&csv))) {
+		deployments = append(deployments, deployment.GetName())
+	}
+
+	operatorCondition := operatorsv1.OperatorCondition{
+		ObjectMeta: metav1.ObjectMeta{
+			// For now, only generate an OperatorCondition with the same name as the csv.
+			Name:            csv.GetName(),
+			Namespace:       csv.Namespace,
+			Labels:          map[string]string{labelKey: ""},
+			OwnerReferences: []metav1.OwnerReference{ownerutil.NonBlockingOwner(&csv)},
+		},
+		Spec: operatorsv1.OperatorConditionSpec{
+			ServiceAccounts: serviceAccounts,
+			Deployments:     deployments,
+		},
+	}
+
+	err = r.Client.Create(context.TODO(), &operatorCondition)
```

Comment:
this should only create/update if it doesn't already exist, right? 

also - does every operator automatically get an OperatorCondition, even if nothing is configured to write to it?

Reply:
>this should only create/update if it doesn't already exist, right?

Yes - but the code will never reach this point because of an earlier[ if statement:](https://github.com/operator-framework/operator-lifecycle-manager/pull/1875/files#diff-2827b0191f1f690c9de2ac31e63e2f3a3b087392633a4ac555f0cd9bfe357ac9R91).

>also - does every operator automatically get an OperatorCondition, even if nothing is configured to write to it?

Yes.